### PR TITLE
Closes #1500 Feature display mode for the news feature

### LINKED ISSUE #1103: Add News view to match C/A "Content Hub" mockups
## Motivation
Casual Astronaut frequently includes a "Content Hub" section in their mockups. This commonly includes a category filter, custom view title, featured article, and category tags on the news teasers (separate issue).

### Screenshot
image

## Solution
- [ ] Add a new view to News that includes a category filter for existing taxonomy
- [ ] #1500
- [ ] Add new view to News using the Feature display mode
- [ ] Connect the view displays into one /content-hub page

### LINKED ISSUE #1500: Add new Feature display mode to News content type
Add new display mode to News to allow for manual embed of new style (view addressed separately)

### PR DESCRIPTION
**Pull request by cpoff**
Closes #1500 

## Description
This adds the News display mode for Content Hub - Feature. Next up will be the the view update(s) separately.
```
core.entity_view_mode.node.feature.yml
core.entity_view_display.node.az_news.feature.yml
```

## Related issues
Closes #1500

## Types of changes

Changes to the display mode that allow for a 'feature article' presentation.

### Arizona Quickstart (install profile, custom modules, custom theme)
- Patch release changes
   - [ ] Bug fix
   - [ ] Accessibility, performance, or security improvement
   - [ ] Critical institutional link or brand change
- Minor release changes
   - [x] New feature
   - [ ] Breaking or visual change to existing behavior
   - [ ] Non-critical brand change
   - [ ] New internal API or API improvement with backwards compatibility
   - [ ] Risky or disruptive cleanup to comply with coding standards
   - [ ] High-risk or disruptive change (requires upgrade path, risks regression, etc.)
- [ ] Other or unknown

### Drupal core
- Patch release changes
   - [ ] Security update
   - [ ] Patch level release (non-security bug-fix release)
   - [ ] Patch removal that's no longer necessary
- Minor release changes
   - [ ] Major or minor level update
- [ ] Other or unknown

### Drupal contrib projects
- Patch release changes
   - [ ] Security update
   - [ ] Patch or minor level update
   - [ ] Add new module
   - [ ] Patch removal that's no longer necessary
- Minor release changes
   - [ ] Major level update
- [ ] Other or unknown

## Checklist

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by cpoff** [comment]
Ready for review. Two errors can be ignored (I believe):

1. Dependency checks
2. PHPStan threw an error, something about theme not specified. Never seen before and remaining test steps went off without a hitch.

**Comment by joeparsons** [comment]
The failed checks should go away if you merge the main branch into your PR branch now.

**Review by joeparsons — COMMENTED** [review]
We need to  prefix the ID / machine name of the node `feature` view mode with `az_` to be consistent with our other view modes.

So the following files need to be renamed: 
```
modules/custom/az_news/config/install/core.entity_view_mode.node.feature.yml
modules/custom/az_news/config/install/core.entity_view_display.node.az_news.feature.yml
```
To:
```
modules/custom/az_news/config/install/core.entity_view_mode.node.az_feature.yml
modules/custom/az_news/config/install/core.entity_view_display.node.az_news.az_feature.yml
```
And then all instances of the following strings:
```
id: node.az_news.feature
core.entity_view_mode.node.feature
mode: feature
view_mode: feature
id: node.feature
```
Need to be updated to:
```
id: node.az_news.az_feature
core.entity_view_mode.node.az_feature
mode: az_feature
view_mode: az_feature
id: node.az_feature
```

**Comment by cpoff** [comment]
@joeparsons Those changes have been committed. The same PHPStan errors persist, but I'm not going to sweat it:

```
13     Drupal\Tests\BrowserTestBase::$defaultTheme is required. See         
19         https://www.drupal.org/node/3083055, which includes recommendations  
20         on which theme to use.  
```

**Comment by camikazegreen** [comment]
The first time I tried to add this view to the homepage, I was unable to save it and "the website encountered an unexpected error".

Here were my settings:
Screen Shot 2022-04-27 at 2 59 45 PM

I tried again and it worked, so it may have just been a probo issue?

**Comment by camikazegreen** [comment]
It can now be viewed on the homepage of Probo here: https://a6aeb03d-8e1b-4710-b048-ecf1607f7bc5.probo.build/

I only saw two new View options available for News though: 

- Content Hub - Feature
- Grid View with Filter

Was there supposed to be a third?

**Comment by cpoff** [comment]
No. Two new view types. The Feature, and the Grid View is for the following news articles. It works in tandem with the feature article and excludes the first news post from the feed.

**Comment by danahertzberg** [comment]
@cpoff Let's do this in more manageable pieces. Please make this PR (or a new PR) close only issue #1500 -- Adding the News display mode. Once that is merged, we will address the view update(s) separately.

**Comment by camikazegreen** [comment]
I think we need to raise the default number of items to either 9, 12, or 15. Hardly need a category filter for three news articles. Open to suggestions on recommended defaults.

I see that we can override it already, which is great!

**Review by trackleft — COMMENTED** [review]
Looks good

**Review by trackleft — DISMISSED** [review]

**Comment by trackleft** [comment]
See @smayden's comment https://github.com/az-digital/az_quickstart/issues/1500#issuecomment-1113565170

File: modules/custom/az_news/config/install/core.entity_view_display.node.az_news.az_feature.yml [hunk 1]
```diff
@@ -0,0 +1,264 @@
...
+    - node.type.az_news
+  module:
+    - date_ap_style
+    - field_group
+    - smart_title
+    - text
+    - user
+third_party_settings:
+  field_group:
+    group_row:
```
Comment (trackleft) [inline modules/custom/az_news/config/install/core.entity_view_display.node.az_news.az_feature.yml:28]: `group_row` is an `h3` but should be a `div` and have the `row` class.

Comment (cpoff) [reply modules/custom/az_news/config/install/core.entity_view_display.node.az_news.az_feature.yml:28]: Ok. That's a two-column layout (first for text, second for the image) so we can use ml-2  on the first col. Or maybe offset-2?

Comment (danahertzberg) [reply modules/custom/az_news/config/install/core.entity_view_display.node.az_news.az_feature.yml:28]: I'm not seeing the row div in the probo build at all. Is this related?

Comment (trackleft) [reply modules/custom/az_news/config/install/core.entity_view_display.node.az_news.az_feature.yml:28]: I'll rebuild

Comment (cpoff) [reply modules/custom/az_news/config/install/core.entity_view_display.node.az_news.az_feature.yml:28]: It's the missing row we identified yesterday (and I thought I added), but it's not in the latest commit. Working to fix now.

Comment (danahertzberg) [reply modules/custom/az_news/config/install/core.entity_view_display.node.az_news.az_feature.yml:28]: Thanks for rebuilding @trackleft! It should pop up after that @cpoff 🤞

**Comment by trackleft** [comment]
This may or may not be important, my preference would be to have the machine name match what the row actually does.

For example: if the field_group is intended to be a bootstrap grid row, I believe that the machine name should be labeled `group_row`
Or if a field group is intended to hold card classes we should use the machine name `group_card`

In the end it probably isn't super important, but it does help with review and it does help with overall neatness of the project.

**Comment by danahertzberg** [comment]
How do we get the Feature display mode to be an option in this list when embedding content via CKEditor?
image

**Comment by trackleft** [comment]
> How do we get the Feature display mode to be an option in this list when embedding content via CKEditor? image

We have to add it to the eligable view modes in ckeditor settings, however, this might backfire on us, since we don't have config in place for other content types for this view mode.  I could be wrong about the backfire though, it is just a guess, and untested.

**Comment by danahertzberg** [comment]
> We'd have to add it to the eligable view modes in ~ckeditor~ entity embed settings, however, this might backfire on us, since we don't have config in place for other content types for this view mode. I could be wrong about the backfire though, it is just a guess, and untested.

I tried it out on my own, and I'm optimistic we can just update the embed.button.az_embed_content.yml file
```
langcode: en
status: true
dependencies:
  config:
    - core.entity_view_mode.node.az_card
    - core.entity_view_mode.node.az_feature
    - core.entity_view_mode.node.az_media_list
    - core.entity_view_mode.node.full
    - core.entity_view_mode.node.teaser
  module:
    - entity_embed
    - node
label: 'Embed Content'
id: az_embed_content
type_id: entity
type_settings:
  entity_type: node
  bundles: {  }
  display_plugins:
    - 'view_mode:node.az_card'
    - 'view_mode:node.az_feature'
    - 'view_mode:node.az_media_list'
    - 'view_mode:node.full'
    - 'view_mode:node.teaser'
  entity_browser: ''
  entity_browser_settings: {  }
icon:
  data: iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAACXBIWXMAAA7EAAAOxAGVKw4bAAAAz0lEQVRYhe2W4Q2CMBBGn8YxGETDGriHi5iwB65B0kHYA/8UbbAHPW0FtC+5cJBre/S+Ngf/zi7h3AVwtL4BuoRreamA3lolBe2/ls5aEzh4vonbJXCzT7fmAKXgw4wmeqW5iYeOefzkKkswYICrYi4DnJ33ErhYvwbaUazIsE2NYnEfv3UMG/SiDGJKA5/iakKseWgC7XzICx3PO0JFLBEGkVID2zgFKTUQNYHaWnQWvweyBjajgWRM7UCBvj1zOb07UNuSqVqwMYuXIJO5AwPmbJF9Gh3mAAAAAElFTkSuQmCC
  uri: 'public://embed_buttons/sharp_post_add_black_24dp.png'
icon_uuid: null

**Comment by trackleft** [comment]
> I tried it out on my own, and I'm optimistic we can just update the embed.button.az_embed_content.yml file

Once we do that we should test how other content types look with that view mode, or if that view mode is available on other content types.

**Comment by danahertzberg** [comment]
> Once we do that we should test how other content types look with that view mode, or if that view mode is available on other content types.
> 
> Or determine that my concerns aren't something to worry about. Either way is fine with me

Ah I see what you're talking about. Yes, it's available to other content types and no bueno. Okay no need to address it here then. :)

File: modules/custom/az_news/config/install/core.entity_view_display.node.az_news.az_feature.yml [hunk 2]
```diff
@@ -0,0 +1,264 @@
...
+      format_settings:
+        element: div
+        show_label: false
+        label_element: h3
+        label_element_classes: ''
+        attributes: ''
+        effect: none
+        speed: fast
+        id: ''
+        classes: 'row no-gutters'
```
Comment (danahertzberg) [inline modules/custom/az_news/config/install/core.entity_view_display.node.az_news.az_feature.yml:102]:
```suggestion
        classes: 'row'
```

File: modules/custom/az_news/config/install/core.entity_view_display.node.az_news.az_feature.yml [hunk 3]
```diff
@@ -0,0 +1,264 @@
...
+      format_settings:
+        element: div
+        show_label: false
+        label_element: h3
+        label_element_classes: ''
+        attributes: ''
+        effect: none
+        speed: fast
+        id: ''
+        classes: 'card card-borderless card-clickable mb-6'
```
Comment (danahertzberg) [inline modules/custom/az_news/config/install/core.entity_view_display.node.az_news.az_feature.yml:44]:
```suggestion
        classes: 'card card-borderless card-clickable mb-5'
```

**Comment by trackleft** [comment]
Also, I should have said this earlier, but IMHO we should be adding enforced dependencies to new config files.

```
dependencies:
  enforced:
    module:
      - az_news
```
This way the config will be removed if the module is uninstalled.

File: modules/custom/az_news/config/install/core.entity_view_display.node.az_news.az_feature.yml [hunk 4]
```diff
@@ -0,0 +1,264 @@
+langcode: en
+status: true
+dependencies:
```
Comment (trackleft) [inline modules/custom/az_news/config/install/core.entity_view_display.node.az_news.az_feature.yml:3]:
```suggestion
dependencies:
  enforced:
    module:
      - az_news
```

File: modules/custom/az_news/config/install/core.entity_view_mode.node.az_feature.yml [hunk 5]
```diff
@@ -0,0 +1,9 @@
+langcode: en
+status: true
+dependencies:
```
Comment (trackleft) [inline modules/custom/az_news/config/install/core.entity_view_mode.node.az_feature.yml:3]:
```suggestion
dependencies:
  enforced:
    module:
      - az_core
```

File: modules/custom/az_news/config/install/core.entity_view_display.node.az_news.az_feature.yml [hunk 6]
```diff
@@ -0,0 +1,264 @@
...
+    - field.field.node.az_news.field_az_short_title
+    - field.field.node.az_news.field_az_subheading
+    - field.field.node.az_news.field_az_summary
+    - node.type.az_news
+  module:
+    - date_ap_style
+    - field_group
+    - smart_title
+    - text
+    - user
```
Comment (trackleft) [inline modules/custom/az_news/config/install/core.entity_view_display.node.az_news.az_feature.yml:25]:
```suggestion
  module:
    - az_core
    - date_ap_style
    - field_group
    - smart_title
    - text
    - user
```

**Comment by cpoff** [comment]
> 

It looks like `field_group` has no other references in the project folder. So I assume we can change to `group_card` without breaking the bank.

**Comment by trackleft** [comment]
`field_group` is the name of the module that adds the third_party_settings to this field, so that has to stay the same.
`group_row` should be changed to `group_card`

**Comment by trackleft** [comment]
It is possible to edit the config file directory to rename field groups, but it is much much easier to do this from within the UI

File: modules/custom/az_news/config/install/core.entity_view_display.node.az_news.az_feature.yml (hunk 4 above)
Comment (trackleft) [reply modules/custom/az_news/config/install/core.entity_view_display.node.az_news.az_feature.yml:3]: @cpoff what do you think about this suggestion?

Comment (trackleft) [reply modules/custom/az_news/config/install/core.entity_view_display.node.az_news.az_feature.yml:3]: Oh, it looks you already applied manually.

Comment (cpoff) [reply modules/custom/az_news/config/install/core.entity_view_display.node.az_news.az_feature.yml:3]: I need to spend some time looking at this a little more, but I'm tied up with Alumni right now.

**Review by joeparsons — APPROVED** [review]

**Review by bberndt-uaz — APPROVED** [review]

File: modules/custom/az_news/config/install/core.entity_view_display.node.az_news.az_feature.yml [hunk 7]
```diff
@@ -17,178 +17,175 @@ dependencies:
     - field.field.node.az_news.field_az_subheading
     - field.field.node.az_news.field_az_summary
     - node.type.az_news
-  enforced:
```
Comment (trackleft) [inline modules/custom/az_news/config/install/core.entity_view_display.node.az_news.az_feature.yml]: We want to keep this.

File: modules/custom/az_news/config/install/core.entity_view_display.node.az_news.az_feature.yml [hunk 8]
```diff
@@ -0,0 +1,247 @@
...
+    - field.field.node.az_news.field_az_short_title
+    - field.field.node.az_news.field_az_subheading
+    - field.field.node.az_news.field_az_summary
+    - node.type.az_news
+  module:
+    - date_ap_style
+    - field_group
+    - smart_title
+    - text
+    - user
```
Comment (trackleft) [inline modules/custom/az_news/config/install/core.entity_view_display.node.az_news.az_feature.yml:25]:
```suggestion
dependencies:
  enforced:
    module:
      - az_core
  config:
    - core.entity_view_mode.node.az_feature
    - field.field.node.az_news.field_az_attachments
    - field.field.node.az_news.field_az_body
    - field.field.node.az_news.field_az_byline
    - field.field.node.az_news.field_az_contacts
    - field.field.node.az_news.field_az_expiration_date
    - field.field.node.az_news.field_az_link
    - field.field.node.az_news.field_az_main_content
    - field.field.node.az_news.field_az_media_image
    - field.field.node.az_news.field_az_news_tags
    - field.field.node.az_news.field_az_published
    - field.field.node.az_news.field_az_short_title
    - field.field.node.az_news.field_az_subheading
    - field.field.node.az_news.field_az_summary
    - node.type.az_news
  module:
    - date_ap_style
    - field_group
    - smart_title
    - text
    - user
```

**Review by trackleft — APPROVED** [review]

**Review by joeparsons — APPROVED** [review]

**Pull request merged**